android/source/src/main/kotlin/com/gu/source/components/pager/PagerProgressIndicator.kt

```diff
@@ -0,0 +1,216 @@
+        indicatorSpacing * (adjustedMaxItems - 1)
+
+    val widthInPx = with(LocalDensity.current) { selectedItemSize.toPx() }
+    LaunchedEffect(pagerState.currentPage) {
+        val viewportSize = listState.layoutInfo.viewportSize
+
+        listState.animateScrollToItem(
+            index = pagerState.currentPage,
+            scrollOffset = (widthInPx / 2 - viewportSize.width / 2).toInt(),
+        )
+    }
+
+    LazyRow(
+        state = listState,
+        modifier = modifier.width(indicatorWidth),
+        horizontalArrangement = Arrangement.spacedBy(indicatorSpacing),
+        verticalAlignment = itemsVerticalAlignment,
+        userScrollEnabled = false,
+    ) {
+        items(pagerState.pageCount) { pageIndex ->
+            PagerProgressItem(
+                itemIndex = pageIndex,
+                selectedItemIndex = pagerState.currentPage,
+                selectedItemSize = selectedItemSize,
+                selectedColour = selectedIndicatorColour,
+                unSelectedColour = unSelectedIndicatorColour,
+                unselectedItemScaleFactor = unselectedItemScaleFactor,
+                itemShape = indicatorShape,
+                numberOfItemsToScale = numberOfItemsToScale.toOddUnder(),
+            )
+        }
+    }
+}
+
+/**
+ * If the number is an even number, return the odd number one below or return the number itself.
+ */
+private fun Int.toOddUnder() = if (this % 2 == 0) this - 1 else this
+
+@Preview
+@Composable
+private fun AnimatedPreview() {
+    val pagerState = rememberPagerState(0) { 10 }
+
+    LaunchedEffect(pagerState) {
+        while (true) {
+            delay(1000)
+            pagerState.scrollToPage((pagerState.currentPage + 1) % pagerState.pageCount)
+        }
+    }
+
+    Column(modifier = Modifier.padding(8.dp)) {
+        HorizontalPager(state = pagerState) {
+            Box(
+                modifier = Modifier
+                    .size(400.dp)
+                    .background(Source.Palette.Neutral86, RoundedCornerShape(8.dp)),
+            ) {
+                Text(
+                    text = (it + 1).toString(),
+                    style = Source.Typography.Titlepiece70,
+                    modifier = Modifier.align(Alignment.Center),
+                    color = Source.Palette.Neutral46,
+                )
+            }
+        }
+        PagerProgressIndicator(
+            pagerState = pagerState,
+            selectedIndicatorColour = Source.Palette.Sport500,
+            unSelectedIndicatorColour = Source.Palette.Neutral73,
+            modifier = Modifier
+                .padding(top = 8.dp)
+                .align(Alignment.CenterHorizontally),
+        )
+    }
+}
+
+@Preview
+@Composable
+internal fun PagerProgressIndicatorPreview() {
+    Column(
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        modifier = Modifier
+            .background(Color.White)
+            .padding(8.dp),
+    ) {
+        Column(modifier = Modifier.padding(8.dp)) {
```

Comment:
Could we use a `repeat { index ->` here to avoid the need to copy paste this 5 times?